api/src/main/java/org/openmrs/module/fhir/api/impl/base/BaseConditionImpl.java

```
@@ -0,0 +1,74 @@
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.fhir.api.impl.base;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.Obs;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir.api.util.FHIRConditionUtil;
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BaseConditionImpl {
+
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#getCondition(String)
+	 */
+	public Condition getCondition(String uuid) {
+		int[] conceptsAsConditions = FHIRUtils.getConceptIdsOfConditions();
+		Obs obs = Context.getObsService().getObsByUuid(uuid);
+		if (obs == null || obs.isVoided() || conceptsAsConditions == null || !ArrayUtils.contains
+				(conceptsAsConditions, obs.getConcept().getId())) {
+			return null;
+		}
+
+		return FHIRConditionUtil.generateFHIRConditionFromOpenMRSObs(obs);
+	}
+
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#createCondition(Condition)
+	 */
+
+	public Condition createCondition(Condition condition) {
+		List<String> errors = new ArrayList<>();
+		if (condition.isEmpty()) {
+			errors.add("Condition resource is null");
+			return null;
+		}
+		if (condition.getSubject().isEmpty()) {
+			errors.add("No subject");
+			return null;
+		}
+		Obs obs = Context.getObsService()
+				.saveObs(FHIRConditionUtil.generateOpenMrsObsFromFHIRCondition(condition), "");
+
+		FHIRUtils.checkGeneratorErrorList(errors);
+
+		return FHIRConditionUtil.generateFHIRConditionFromOpenMRSObs(obs);
+	}
+
+
+	public org.openmrs.Condition generateOpenMrsCondition(Condition condition) throws NotImplementedException {
+		return null;
```

Can we throw NotImplementedException instead of returning null?